src/root.component.tsx

```diff
@@ -6,6 +6,44 @@ import PatientBanner from "./summary/banner/patient-banner.component";
 import LevelTwoRoutes from "./summary/level-two-routes.component";
 
 function Root(props) {
+  const [widgetRoutes, setWidgetRoutes] = React.useState([]);
+
+  const exampleConfig = [
+    {
+      name: "Programs",
+      esModule: "@jj-widgets",
+      exportName: "programsWidget"
+    }
+  ];
+
+  const config = [];
+
+  React.useEffect(() => {
+    const modulePromises = [];
+
+    config.map(c => {
+      if (c["esModule"]) {
+        modulePromises.push(System.import(c.esModule));
+      }
+    });
+
+    //@ts-ignore
+    Promise.allSettled(modulePromises).then(modules => {
+      const importedWidgets = [];
+      let widgetRoutes = [];
+      modules.map(m => {
+        if (m.status === "fulfilled") {
+          for (let [exportName, widget] of Object.entries(m.value.widgets)) {
+            if (widget.hasOwnProperty("routes")) {
+              widgetRoutes.push(widget["routes"]);
+            }
+          }
+        }
+      });
+      setWidgetRoutes(widgetRoutes);
+    });
```

Comment:
What happens if one rejects?

Reply:
Good question. I'm not sure what to do. We could (1) do nothing; (2) warn the user that some data couldn't load; (3) try to reimport; (4) not show anything. I don't know what the best option is but I would lean towards (2). However, I'm not sure how a user would interpret this information but I think it's the best option to investigate. What do you think?

Reply:
Agree on (2). Also is consistent with our previous decision about errors and honesty.

Reply:
I've updated such that we create a toast for each module/widget that is unable to be loaded.